src/cryptoadvance/specter/util/bitcoind_setup_tasks.py

```diff
@@ -0,0 +1,212 @@
+import os, time, requests, secrets, platform, tarfile, zipfile, sys
+from ..bitcoind import BitcoindPlainController
+import pgpy
+from pathlib import Path
+from .sha256sum import sha256sum
+
+
+def setup_bitcoind_thread(specter=None):
+    try:
+        BITCOIND_OS_SUFFIX = {
+            "Windows": "win64.zip",
+            "Linux": "x86_64-linux-gnu.tar.gz",
+            "Darwin": "osx64.tar.gz",
+        }
+        bitcoind_url = f"https://bitcoincore.org/bin/bitcoin-core-0.21.0/bitcoin-0.21.0-{BITCOIND_OS_SUFFIX[platform.system()]}"
+        response = specter.requests_session().get(bitcoind_url, stream=True)
+        packed_name = os.path.join(
+            specter.data_folder,
+            f"bitcoind-{BITCOIND_OS_SUFFIX[platform.system()]}",
+        )
+        with open(packed_name, "wb") as f:
+            total_length = float(response.headers["content-length"])
+            downloaded = 0.0
+            old_progress = 0
+            specter.config["bitcoind_setup"][
+                "stage"
+            ] = "Downloading Bitcoin Core release files..."
+            specter._save()
+            for chunk in response.iter_content(chunk_size=4096):
+                downloaded += len(chunk)
+                f.write(chunk)
+                new_progress = int((downloaded / total_length) * 10000) / 100
+                if new_progress > old_progress:
+                    old_progress = new_progress
+                    specter.config["bitcoind_setup"]["stage_progress"] = new_progress
+                    specter._save()
+        specter.config["bitcoind_setup"]["stage"] = "Verifying signatures..."
+        specter._save()
+        bitcoind_sha256sums_url = (
+            "https://bitcoincore.org/bin/bitcoin-core-0.21.0/SHA256SUMS.asc"
+        )
+        response = specter.requests_session().get(bitcoind_sha256sums_url, stream=True)
+        bitcoind_sha256sums_file = os.path.join(
+            specter.data_folder, "bitcoind-sha256sums.asc"
+        )
+        with open(bitcoind_sha256sums_file, "wb") as f:
+            total_length = float(response.headers["content-length"])
+            for chunk in response.iter_content(chunk_size=4096):
+                f.write(chunk)
+        with open(bitcoind_sha256sums_file, "r") as f:
+            signed_sums = f.read()
+            bitcoind_release_pgp_key, _ = pgpy.PGPKey.from_file(
+                os.path.join(sys._MEIPASS, "static/bitcoin-release-pubkey.asc")
+                if getattr(sys, "frozen", False)
+                else Path(__file__).parent / "../static/bitcoin-release-pubkey.asc"
+            )
+            bitcoind_sha256sums_msg = pgpy.PGPMessage.from_file(
+                bitcoind_sha256sums_file
+            )
+            if not bitcoind_release_pgp_key.verify(bitcoind_sha256sums_msg):
+                raise Exception("Failed to verify Bitcoin Core PGP signature")
```

Comment:
We have to be aware that this will end up in the logs only. Can't we do something like:
```
specter.config["bitcoind_setup"]["stage"] = "Installation failed due to not being able to verify the PGP signature"
specter._save()
```
I guess this could then be shown (maybe with some extra-logic to end the "frontend-polling-mode" ?